### DISCUSSION TITLE
TAWG Vision - What do we mean by Adoption?

### DISCUSSION THREAD

**Discussion by chrisisla** [finos/common-domain-model#2859] [General]
When we talk about firms _adopting_ the CDM, what do we actually mean? 

The purpose of this discussion is to start coming up with a list of elements that we can agree are required in order to state that the CDM has been _adopted_ by a firm/institution.

The TAWG can then come up with technology strategies for how to assist a firm adopt the CDM - without knowing what adoption means, the group cannot come up with strategies to get there.

> [!NOTE]
> _The TAWG is purely focused on the underlying technology of the CDM, not the functionality/content of the model itself. The CRWG and SWG are better placed to come up with functional enhancements to promote adoption_

**Comment by gmackenzie1000** [comment]
I've not been involved in the discussions to date, but here are some thoughts that come to mind:
Some elements that would show that a firm is adopting CDM would be:
- Centre of Excellence - a small but dedicated group that would act as a point of contact for adopters inside the firm and act as a interface to the CDM working groups. 
- Architecture guidelines and standards - the firm's standards incorporating direction for when and how CDM should or must be used. These would be managed by the CoE in collaboration with the broader architecture governance structures
- Roadmap - an actionable plan of adoptions, for example timelines for systems to convert  to CDM or new CDM-based systems to be implemented.

**Comment by iansloyan** [comment]
What is adoption of the CDM?

In-whole or in-part:
- where the code of the CDM, as found in this repository, is being used by a firm in their systems, in the systems provided to them by a vendor, or in the systems of a market infrastructure which they must connect to in their business.
- where a firm is modelling their own data, by adhering to the CDM as a data standard, in their production databases.

Why "in-whole" or "in-part"? 
Well "in-whole" is probably impossible for all, but a very small, or very new firm, that can may use all of the CDM, or all of the CDM that pertains to their specific business; e.g. a brand new sec lending platform that uses all of the CDM model for their product and event model with only the necessary (but still likely the majority of the code!) UI and infrastructural code around it to make the system work. 
"In-part" could mean that in one product line for a specific use case a firm has implemented part of the CDM; e.g. a custodian with existing infrastructure and clients that takes their collateral inventory or eligibility workflow and implements changes to this to use the CDM collateral specification as the representation for communicating and matching collateral with clients.

So where does that leave the work of the TAWG - I think in the space of deployment of the code (so versioning, delivery, release processes, technical formats, code generators, validation and security tools), and the connectivity of the model to existing standards and formats to allow easier transformation of existing data to CDM format for storage or transmission. 

A strong area for the CDM should also be the modularity of the technical architecture, so that things can expand and develop in the model but not radically impact other areas which are much more stable and in production. So enhancements to the product model as demanded by a large cohort of implementors do not impact the collateral specification model which is being used by another separate cohort - namespaces and good versioning practices were always expected to help achieve this but it should not be at the expense of the design principles of Normalisation and Composability (which can mean that changes to base components can have ripples throughout the model).

**Reply by iansloyan** [reply]
I prematurely posted the above (with some keyboard shortcut I cannot replicate) - but hopefully it helps or at least can spark from questions to further define _Adoption_?

**Comment by dschwartznyc** [comment]
perhaps adoption means that a CDM library is available for in-production use in an organization's infrastructure.  at one end of the spectrum, production usage might mean a single application or service where the context (type and CDM version) of data usage is fully known.  at the other end, the application or service might have no direct a priori knowledge of the context of the data.  

while the definition of adoption above might be seen as strict, the route to adoption is very flexible.  it may begin with any form of exploration and engagement (modeling, code downloads, ...).  it may progress to participation in working groups, contributions, et al. regardless, it would seem the true end goal is that the number of organizations using the standard grows.

**Comment by chrisisla** [comment]
The topic of Adoption was discussed and agreed at the TAWG meeting on 9th May. Thanks to all who contributed to this discussion, closing this now.
